### PR TITLE
Deploy UDMS code to firebase

### PR DESCRIPTION
**Pull request by anurag6**

### CODE REVIEW

File: udms/e2e/firestore/import_db_cloud.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
+// Point json_data_source to the json file with data to be populated into the DB
+// Point db_collection to the collection name/topmost field of json object being
+// populated.
+// If script fails with permission issues, run:
+// 'gcloud auth application-default login'
+
```
Comment (henry54809) [inline udms/e2e/firestore/import_db_cloud.js:6]: You can import functions from the original script. It looks like the 2 are almost identical.

File: udms/README.md [hunk 2]
```diff
@@ -37,6 +37,12 @@ Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.
 
 Run `ng e2e` to execute the end-to-end tests via [Protractor](http://www.protractortest.org/).
 
+## Deploying
+
+To populate the firestore database, run `node e2e/firestore/import_db_cloud.js`.
+
+To deploy, build the project using `ng build --prod` and deploy using `firebase deploy.
```
Comment (grafnu) [inline udms/README.md:44]: missing closing quote after deploy.

File: udms/angular.json [hunk 3]
```diff
@@ -44,7 +44,14 @@
                   "with": "src/environments/environment.prod.ts"
                 }
               ],
-              "optimization": true,
+	      "optimization": {
```
Comment (grafnu) [inline udms/angular.json:47]: indent seems incongruous with rest of file

File: udms/e2e/firestore/import_db_cloud.js (hunk 1 above)
Comment (anurag6) [reply udms/e2e/firestore/import_db_cloud.js:10]: Changed.

File: udms/README.md (hunk 2 above)
Comment (anurag6) [reply udms/README.md]: Changed.

File: udms/angular.json (hunk 3 above)
Comment (anurag6) [reply udms/angular.json]: vimrc was configured in a weird way which messed up indents. Corrected.

**Review by grafnu — APPROVED** [review]

File: udms/e2e/firestore/import_db.js [hunk 4]
```diff
@@ -55,7 +49,7 @@ async function importDoc(ref, doc) {
         } else if (isCollection(value)) {
             collections.push(key);
         } else if (isMap(value)) {
-            formattedDoc[key] = formatMap(value) 
+            formattedDoc[key] = formatMap(value)
```
Comment (henry54809) [inline udms/e2e/firestore/import_db.js:52]: There should be a ; there.

File: udms/e2e/firestore/import_db_cloud.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+// Point json_data_source to the json file with data to be populated into the DB
+// Point db_collection to the collection name/topmost field of json object being
+// populated.
+// If script fails with permission issues, run:
+// 'gcloud auth application-default login'
+
+const firebase = require("firebase-admin");
+require("firebase/firestore");
```
Comment (henry54809) [inline udms/e2e/firestore/import_db_cloud.js:12]: Is this required necessary? Seems like it’s used anywhere.

**Review by henry54809 — COMMENTED** [review]
Just 2 small things.

File: udms/e2e/firestore/import_db_cloud.js (hunk 5 above)
Comment (anurag6) [reply udms/e2e/firestore/import_db_cloud.js]: Removed.

File: udms/e2e/firestore/import_db.js (hunk 4 above)
Comment (anurag6) [reply udms/e2e/firestore/import_db.js]: Changed

**Review by henry54809 — APPROVED** [review]

**Pull request merged**